src/revlanguage/functions/phylogenetics/ratematrix/Func_revPoMo2N.cpp

```diff
@@ -0,0 +1,121 @@
+#include "ModelVector.h"
+#include "Natural.h"
+#include "Real.h"
+#include "RealPos.h"
+#include "RlDeterministicNode.h"
+#include "RlRateMatrix.h"
+#include "RlSimplex.h"
+#include "TypedDagNode.h"
+
+using namespace RevLanguage;
+
+/** default constructor */
+Func_revPoMo2N::Func_revPoMo2N( void ) : TypedFunction<RateMatrix>( )
+{
+    
+}
+
+
+/**
+ * The clone function is a convenience function to create proper copies of inherited objected.
+ * E.g. a.clone() will create a clone of the correct type even if 'a' is of derived type 'b'.
+ *
+ * \return A new copy of the process.
+ */
+Func_revPoMo2N* Func_revPoMo2N::clone( void ) const
+{
+    
+    return new Func_revPoMo2N( *this );
+}
+
+
+RevBayesCore::TypedFunction< RevBayesCore::RateGenerator >* Func_revPoMo2N::createFunction( void ) const
+{
+    RevBayesCore::TypedDagNode< long                          >* ni = static_cast<const Natural              &>( this->args[0].getVariable()->getRevObject() ).getDagNode();
+    RevBayesCore::TypedDagNode<RevBayesCore::Simplex          >* bf = static_cast<const Simplex              &>( this->args[1].getVariable()->getRevObject() ).getDagNode();
+    RevBayesCore::TypedDagNode<RevBayesCore::RbVector<double> >* ex = static_cast<const ModelVector<RealPos> &>( this->args[2].getVariable()->getRevObject() ).getDagNode();
+    RevBayesCore::TypedDagNode<RevBayesCore::RbVector<double> >* fc = static_cast<const ModelVector<RealPos> &>( this->args[3].getVariable()->getRevObject() ).getDagNode();
+
+    if ( ex->getValue().size() != 1 )
+    {
+        throw RbException("The number of exchangeabilities should match the number of biallelic polymorphic types: 1.");
+    }
+    if ( bf->getValue().size() != 2 )
+    {
+        throw RbException("The number of allele frequencies should match the number of alleles: 2.");
+    }
+    if ( fc->getValue().size() != 2 )
+    {
+        throw RbException("The number of fitness coeffiients should match the number of alleles: 2.");
+    }
+
+    RevBayesCore::revPoMo2NRateMatrixFunction* f = new RevBayesCore::revPoMo2NRateMatrixFunction( ni, bf, ex, fc );
+    
+    return f;
+}
+
+
+/* Get argument rules */
+const ArgumentRules& Func_revPoMo2N::getArgumentRules( void ) const
+{
+    
+    static ArgumentRules argumentRules = ArgumentRules();
+    static bool          rules_set = false;
+    
+    if ( !rules_set )
+    {
+        argumentRules.push_back( new ArgumentRule( "N"   , Natural::getClassTypeSpec(), "Number of individuals.", ArgumentRule::BY_CONSTANT_REFERENCE, ArgumentRule::ANY ) );
+        argumentRules.push_back( new ArgumentRule( "pi"  , Simplex::getClassTypeSpec(), "Vector of allele frequencies: pi=(pi_a0,pi_a1).", ArgumentRule::BY_CONSTANT_REFERENCE, ArgumentRule::ANY ) );
+        argumentRules.push_back( new ArgumentRule( "rho" , ModelVector<RealPos>::getClassTypeSpec(), "Vector of mutation rates: mu=(mu_a0a1,mu_a1a0).", ArgumentRule::BY_CONSTANT_REFERENCE, ArgumentRule::ANY ) );
```

Comment:
is this argument rho or mu ? is it a vector of mutation rates or exchangeabilities ? (l44)

Reply:
updated pomo rate matrices argument rules and commited